.github/workflows/smoke-test/smoke-test.sh

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env bash
+
+# This script is expected to run from CI, and does a final smoke test.
+# There should be an installer in the current directory (or, in the case of
+# Linux, a zip file), along with its accompanying sha512sum file.
+# On Windows, build/signing-config-win.yaml is also required.
+
+# Required tools:
+# - jq
+# - yq (Windows only)
+
+# Note that, on Windows, this is run via msys bash (installed wit git).
+
+set -o errexit -o nounset
+
+# All commands in the cleanups array will be run on exit.  They must be plain
+# strings that will be passed to eval
+cleanups=()
+
+# Run the cleanups.
+do_cleanup() {
+    # In case the array has holes (it shouldn't), make an array of indices.
+    local indices=("${!cleanups[@]}")
+    local i
+    for (( i=${#indices[@]} - 1; i >= 0; i-- )); do
+        # shellcheck disable=2086 # We expect to glob and word split
+        eval ${cleanups[$i]}
+    done
+}
+
+trap do_cleanup EXIT
+
+# Locate the archive, check its checksum, and echo the file name.
+get_archive() {
+    local checksum archiveName
+    for checksum in *.sha512sum; do
+        archiveName=${checksum%.sha512sum}
+        if command -v sha512sum &>/dev/null; then
+            sha512sum --check --quiet --strict "$checksum"
+        else
+            shasum --check --quiet --algorithm 512 "$checksum"
+        fi
+        grep --quiet "$archiveName" "$checksum"
+        readlink -f "$archiveName"
+        return
+    done
+    echo "Failed to find archive." >&2
+    exit 1
+}
+
+# Return the current platform; one of "darwin", "linux", "win32"
+get_platform() {
+    case "$(uname -s)" in
+    Darwin)
+        echo "darwin";;
+    Linux)
+        echo "linux";;
+    MINGW*)
+        echo "win32";;
+    *)
+        printf "Unsupported platform %s\n" "$(uname -s)" >&2
+        exit 1;;
+    esac
+}
+
+# Assume the first argument given is a path to the Rancher Desktop .dmg disk
+# image; install it, and set the global variable RDCTL to the path of the rdctl
+# executable.
+install_darwin() {
+    local archiveName=$1
+    local mountpoint
+    mountpoint=$(mktemp -d -t rd-dmg-)
+    cleanups+=("rm -rf '$mountpoint'")
+
+    local srcApp="${mountpoint}/Rancher Desktop.app"
+    local destApp="/Applications/Rancher Desktop.app"
+
+    codesign --verify --deep --strict --verbose=2 --check-notarization "$archiveName"
+    hdiutil attach "$archiveName" -mountpoint "$mountpoint"
+    cleanups+=("hdiutil detach '$mountpoint'")
+
+    codesign --verify --deep --strict --verbose=2 --check-notarization "$srcApp"
+    mkdir -p "$destApp"
+    cleanups+=("rm -rf '$destApp'")
+
+    tar -cC "$srcApp" . | tar -xC "$destApp"
```

Comment:
This looks more complicated than `cp -R …`. The only reason I can think of where `tar` is better is if the source can contain hard links, which should not be the case here. Are there other reasons?